### PR TITLE
chore(papyrus_p2p_sync): add run_test action for starting p2p_sync

### PR DESCRIPTION
**Pull request by eitanm-starkware**

### CODE REVIEW

**Review by ShahakShama — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2840)* status: all files reviewed, 4 unresolved discussions (waiting on @eitanm-starkware)

___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 161 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OEhwwOi7mFPs-mAq010:-OEhwwOjBrdbc62lfgVT:b-wdqwqd) ([raw file](https://github.com/starkware-libs/sequencer/blob/1f863cb7261e0160e2d5dd339dd7fd2109e204ea/crates/papyrus_p2p_sync/src/client/test_utils.rs#L161)):*
> ```XML
>     ValidateReportSent(DataType),
>     /// Run the P2P sync client.
>     RunP2pSync,
> ```

Put this action at the front
___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 212 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OEhx4jG7U_v298CjXA3:-OEhx4jG7U_v298CjXA4:bnd7oil) ([raw file](https://github.com/starkware-libs/sequencer/blob/1f863cb7261e0160e2d5dd339dd7fd2109e204ea/crates/papyrus_p2p_sync/src/client/test_utils.rs#L212)):*
> ```XML
>     let mut transaction_current_query_responses_manager = None;
>     let mut class_current_query_responses_manager = None;
>     if !actions.contains(&Action::RunP2pSync) {
> ```

explicit is better than implicit.
Remove this and add manually the RunP2pSync action at the start of every test
___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 303 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OEiDrnE8j7V96bG_Ro1:-OEiDrnE8j7V96bG_Ro2:b-whnhjl) ([raw file](https://github.com/starkware-libs/sequencer/blob/1f863cb7261e0160e2d5dd339dd7fd2109e204ea/crates/papyrus_p2p_sync/src/client/test_utils.rs#L303)):*
> ```XML
>                     Action::RunP2pSync => {
>                         let p2p_sync = p2p_sync.take().expect("Called RunP2pSync twice");
>                         tokio::spawn(
> ```

store the handle returned and poll it in the select
You can have a variable that holds a boxed future and starts with pending() and is changed here.

**Review by eitanm-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2840)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @graphite-app[bot] and @ShahakShama)

___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 212 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OEhx4jG7U_v298CjXA3:-OEnUclP16o0bDie4ub2:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/1f863cb7261e0160e2d5dd339dd7fd2109e204ea/crates/papyrus_p2p_sync/src/client/test_utils.rs#L212)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

explicit is better than implicit.
Remove this and add manually the RunP2pSync action at the start of every test
</blockquote></details>

Done.
___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 303 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OEiDrnE8j7V96bG_Ro1:-OEnUde_89RK9LFUQOs3:bdn1jux) ([raw file](https://github.com/starkware-libs/sequencer/blob/1f863cb7261e0160e2d5dd339dd7fd2109e204ea/crates/papyrus_p2p_sync/src/client/test_utils.rs#L303)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

store the handle returned and poll it in the select
You can have a variable that holds a boxed future and starts with pending() and is changed here.
</blockquote></details>

Done. - used oneshot channels instead

**Review by ShahakShama — CHANGES_REQUESTED** [review]
> Reviewed 1 of 3 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2840)* status: all files reviewed, 4 unresolved discussions (waiting on @eitanm-starkware)

___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 207 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OErz4HB4UuQa2Z7y7tZ:-OErz4HB4UuQa2Z7y7t_:bahblsr) ([raw file](https://github.com/starkware-libs/sequencer/blob/f1a1fb228bc877964ebeaaa6c2d42eb0d379e658/crates/papyrus_p2p_sync/src/client/test_utils.rs#L207)):*
> ```XML
> 
>     // create fut pending poll
>     let (run_p2p_sync_sender, run_p2p_sync_receiver) = oneshot::channel();
> ```

Rename to sync_future_sender and sync_future_receiver
___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 301 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OErzB-i3rq_64aN_F6h:-OErzB-i3rq_64aN_F6i:b-8qctjw) ([raw file](https://github.com/starkware-libs/sequencer/blob/f1a1fb228bc877964ebeaaa6c2d42eb0d379e658/crates/papyrus_p2p_sync/src/client/test_utils.rs#L301)):*
> ```XML
>             }
>         } => {},
>         res = run_p2p_sync_receiver.then(|res| async { res.expect("Failed to run P2P sync");
> ```

Add newline after/before the curly brackets (both the { and the })
___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 304 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OErzfaR2bkL4Q3XoJ4M:-OErzfaR2bkL4Q3XoJ4N:b39sob8) ([raw file](https://github.com/starkware-libs/sequencer/blob/f1a1fb228bc877964ebeaaa6c2d42eb0d379e658/crates/papyrus_p2p_sync/src/client/test_utils.rs#L304)):*
> ```XML
>         p2p_sync.run().await}) => {
>             res.unwrap();
>             panic!("P2P sync client finished running before all actions were executed");
> ```

It should never finish running so you can keep the previous message. I'm trying to push for us to change the return type of run to P2pSyncError instead of Result<(), P2pSyncError> but it's a change we'd better do across the entire repo

File: crates/papyrus_p2p_sync/src/client/test_utils.rs [hunk 1]
```diff
@@ -156,10 +157,18 @@ pub enum Action {
...
     ValidateReportSent(DataType),
+    /// Run the P2P sync client.
+    RunP2pSync,
+}
+
+impl PartialEq for Action {
+    fn eq(&self, other: &Self) -> bool {
+        matches!((self, other), (Action::RunP2pSync, Action::RunP2pSync))
+    }
 }
```
[the remark this answers is not in the document]
Comment (eitanm-starkware) [reply crates/papyrus_p2p_sync/src/client/test_utils.rs]: Done.

**Review by eitanm-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2840)* status: all files reviewed, 4 unresolved discussions (waiting on @graphite-app[bot] and @ShahakShama)

___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 207 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OErz4HB4UuQa2Z7y7tZ:-OEs1vwWCDdWQ3B958AV:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/f1a1fb228bc877964ebeaaa6c2d42eb0d379e658/crates/papyrus_p2p_sync/src/client/test_utils.rs#L207)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Rename to sync_future_sender and sync_future_receiver
</blockquote></details>

Done.
___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 301 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OErzB-i3rq_64aN_F6h:-OEs2NYPAoanyX2yMXoP:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/f1a1fb228bc877964ebeaaa6c2d42eb0d379e658/crates/papyrus_p2p_sync/src/client/test_utils.rs#L301)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Add newline after/before the curly brackets (both the { and the })
</blockquote></details>

Done.
___
*[`crates/papyrus_p2p_sync/src/client/test_utils.rs` line 304 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-OErzfaR2bkL4Q3XoJ4M:-OEs2gc412o4ytUcL6bN:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/f1a1fb228bc877964ebeaaa6c2d42eb0d379e658/crates/papyrus_p2p_sync/src/client/test_utils.rs#L304)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

It should never finish running so you can keep the previous message. I'm trying to push for us to change the return type of run to P2pSyncError instead of Result<(), P2pSyncError> but it's a change we'd better do across the entire repo
</blockquote></details>

Done.

**Review by ShahakShama — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2840#-:-OEsYIxr4UlyQASiJgdX:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @graphite-app[bot])

**Review by eitanm-starkware — COMMENTED** [review]
> Dismissed @graphite-app[bot] from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2840)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @eitanm-starkware)

**Pull request merged**